src/javascript/binary/pages/client/change_passwordws.js

```diff
@@ -0,0 +1,144 @@
+var PasswordWS = (function(){
+
+	var $form, $result;
+
+	var init = function() {
+		$form   = $("#change-password > form");
+		$result = $("#change-password > div[data-id='success-result']");
+		$form.find("button").on("click", function(e){
+			e.preventDefault();
+			e.stopPropagation();
+			PasswordWS.sendRequest();
+		});
+	};
+	
+	var validateForm = function() {
+
+		var isValid 	= true,
+			old_pass 	= $form.find("input[name='oldpassword']").val(),
+			new_pass 	= $form.find("input[name='new-password']").val(),
+			repeat_pass = $form.find("input[name='repeat-password']").val();
+
+		/**
+		 * Validation for new-password
+		**/
+
+		// Old passwrod cannot be blank. We leave the actual matching to backend
+		if(0 === old_pass.length) {
+			$form.find("p[data-error='old-blank']").removeClass("hidden");
+			isValid = false;
+		} else {
+			$form.find("p[data-error='old-blank']").addClass("hidden");
+		}		
+
+		// New password cannot be the same as the old password
+		if(new_pass.length > 0 && new_pass === old_pass) {
+			$form.find("p[data-error='same-as-old']").removeClass("hidden");
+			isValid = false;
+		} else {
+			$form.find("p[data-error='same-as-old']").addClass("hidden");
+		}
+
+		// Min length
+		if(new_pass.length < 6) {
+			$form.find("p[data-error='too-short']").removeClass("hidden");
+			isValid = false;
+		} else {
+			$form.find("p[data-error='too-short']").addClass("hidden");
+		}
+
+		// Max length
+		if(new_pass.length > 25) {
+			$form.find("p[data-error='too-long']").removeClass("hidden");
+			isValid = false;
+		} else {
+			$form.find("p[data-error='too-long']").addClass("hidden");
+		}
+
+		// Invalid characters
+		var regexp = new RegExp('^[ -~]+$');
```

Comment:
The API should sanitize the input. Restricting characters for passwords is a known security anti-pattern.